Reproduce and fix Siythrun's issue from #388

gdubicki opened the pull request.

gdubicki committed:
Fix Siythrun's issue from #388 + refactor transform tests

by following wwkimball's advice from
https://github.com/gitlabform/gitlabform/pull/467#discussion_r1064063152

gdubicki commented:
I got stuck and reported an issue to yamlpath library author here: https://github.com/wwkimball/yamlpath/issues/206

wwkimball commented on gitlabform/configuration/transform.py:
```diff
@@ -39,82 +41,92 @@ def transform(self, configuration: Configuration) -> None:
-                # under the given path
-                pass
+        try:
+            for node_coordinate in processor.get_nodes(
```
When querying data -- especially when using wildcards -- be sure to set `mustexist=True` as shown in [the YAML Path documentation](https://github.com/wwkimball/yamlpath/wiki/Python-Library#getting-data-from-the-document) which illustrates how to get data out of the document.  Leaving this flag at the default will cause the engine to attempt to generate any missing path ancestry, which is unknown where wildcards are used.

gdubicki committed:
Correct sub_group -> group name + cleanups

gdubicki committed:
The user need to be a member to be made an approver

and the make_user() only made them a member of group_and_project,
NOT project_in_subgroup.

This issue made me realize what is the issue:
https://gitlab.com/gitlab-org/gitlab/-/issues/324302
